hazelcast/src/main/java/com/hazelcast/map/impl/eviction/ExpirationManager.java

```diff
@@ -132,21 +133,31 @@
     public static final HazelcastProperty CLEANUP_OPERATION_COUNT
             = new HazelcastProperty(PROP_CLEANUP_OPERATION_COUNT);
 
-    private final boolean primaryDrivesEviction;
-    private final int taskPeriodSeconds;
-    private final int partitionCount;
-    private final int cleanupPercentage;
-    private final int cleanupOperationCount;
-    private final Address thisAddress;
-    private final NodeEngine nodeEngine;
-    private final ClusterService clusterService;
-    private final HazelcastProperties properties;
-    private final ExecutionService executionService;
-    private final IPartitionService partitionService;
-    private final InternalOperationService operationService;
-    private final PartitionContainer[] partitionContainers;
-
-    private ScheduledFuture<?> expirationTask;
+    final boolean primaryDrivesEviction;
```

Comment:
I'm not a huge fan of this exposure, since you could alter the state e.g. of the `scheduled` flag from the outside. Also from just looking at the code, you cannot directly see if `ExpirationManager` is the only class which is changing these states. With private access this is much easier to understand.

Also tests like `expirationManager.scheduled.get()` are relying on an implementation detail. I would prefer to hide this behind a `expirationManager.isScheduled()` method, so the implementation can be changed without the need to fix tests at the same time.

Reply:
fixed